scripts/Engineering_Diffraction.py

```diff
@@ -9,19 +9,20 @@
 from qtpy import QtCore
 import sys
 
-# If the GUI has not been created yet, make a new one.
-if 'engineering_gui' not in globals():
+if 'engineering_gui' in globals():
+    engineering_gui = globals()['engineering_gui']
+    if not engineering_gui.isHidden():
```

Comment:
If I start up workbench and then open up the engineering diffraction UI straight away it doesn't have a minimize button in its title bar. If I then close it down and open it up again it does have a minimize button.
It seems like it should have a minimize button in both cases? I wondered if it related to this if statement where it goes in the second branch when opening on the second time for some reason. isHidden seems to return a different value in the two situations

Re our conversation yesterday I think it's fine for the engineering interface to continue the tracking\management of the log workspace group while it's minimized. It's only when it's been closed that it needs to let go\stop controlling the log workspace group. Maybe that's already clear but thought I'd mention it while we're talking about minimizing the window.

Reply:
I can't recreate this - on my debug build on windows it has a minimise button...what OS are you using? Re the tracking of the log group I agree that this should (and indeed does on my build) persist even when the window is minimised.

Reply:
> I can't recreate this - on my debug build on windows it has a minimise button...what OS are you using? Re the tracking of the log group I agree that this should (and indeed does on my build) persist even when the window is minimised.

I was testing on Ubuntu. I can try and find out what's going on with the isHidden value but might be after lunch because on a course this morning. Sorry for delay...

Reply:
OK thanks no rush, I guess I just need to know what the two values it returns are and compare it to what I get on windows. I promise I'll get my own ubuntu build setup soon!

Reply:
Had a few mins to test and debug on Ubuntu.
So when I first open up the interface the code goes into the second branch of the "if 'engineering_gui' in globals():" statement ie lines 22-28 are executed and it gets the window flags from the get_window_config call
When I close it down and reopen it, the code goes into the first branch of the "if 'engineering_gui' in globals()" statement with isHidden=true. So lines 19 and 20 are executed.

So it seems like the get_window_config call returns some flags with the minimize option disabled on Ubuntu. Maybe minimizing windows isn't great on Ubuntu for some reason. The behaviour is a bit odd - when I minimize a window it just gets hidden behind the main window and it's not obvious where it's gone.

For consistency you should probably also call get_window_config in lines 19-20 somewhere and pass some flags into the EngineeringDiffractionGui constructor? Do you actually know what get_window_config does - I'm not sure?